docs/ir/ir-breaking-changes.rst

```diff
@@ -132,4 +170,32 @@ the right function.
 The ID ``0`` is reserved for uninitialized function pointers which then cause a panic in the disptach function when called.
 
 In the old code generator, internal function pointers are initialized with a special function that always causes a panic.
-This causes a storage write at construction time for internal function pointers in storage.
\ No newline at end of file
+This causes a storage write at construction time for internal function pointers in storage.
+
+Clean up:
+
+The old code generator only performs cleanup before an operation where it is important to have dirty bits cleaned.
+The new code generator performs cleanup after any operation that can result in dirty bits.
+
+For example:
+::
+     // SPDX-License-Identifier: GPL-3.0
+     pragma solidity >0.8.0;
+     contract C {
+         function f(uint8 a) public returns(uint r1, uint r2)
+         {
+             a = ~a;
+             assembly {
+                 r1 := a
+             }
+             r2 = a;
+         }
+     }
+
+The function `f(uint8)` returns the following for the input 1:
```

Comment:
This sounded a bit confusing to me at first until I realized that by "input 1" you mean passing `1` to the function rather than referring to the argument at position 1. Maybe this wording would be better?
```suggestion
The call ``f(1)`` returns the following values:
```